src/main/java/org/gradle/profiler/GradleBuildInvocationResult.java

```diff
@@ -21,9 +24,24 @@ public Duration extractFrom(GradleBuildInvocationResult result) {
         }
     };
 
-    public GradleBuildInvocationResult(String displayName, Duration executionTime, @Nullable Duration timeToTaskExecution, String daemonPid) {
+    public static Sample<GradleBuildInvocationResult> sampleBuildOperation(String buildOperationDetailsClass) {
+        return new Sample<GradleBuildInvocationResult>() {
+            @Override
+            public String getName() {
+                return buildOperationDetailsClass;
+            }
+
+            @Override
+            public Duration extractFrom(GradleBuildInvocationResult result) {
+                return Optional.ofNullable(result.cumulativeBuildOperationTimes.get(buildOperationDetailsClass)).orElse(Duration.ofMillis(0));
```

Comment:
It threw me off a bit that the method should return a `Duration` yet from the start of the `return` statement I expected it to return an `Optional`. If we keep the use of `Optional` here, I think it's important to break `orElse()` to a new line.

OTOH, I don't think this is a good use of `Optional`: we create an object just to be thrown away immediately, and I don't think it's easier to read than this:

```java
Duration duration = result.cumulativeBuildOperationTimes.get(buildOperationDetailsClass);
return duration != null
    ? duration
    : Duration.ofMillis(0);
```

Furthermore, `Duration.ofMillis(0)` should be replaced with `Duration.ZERO`.

Reply:
If allocation is important here, we can check whether escape analysis kick in here - otherwise I would vote for the newline.

Reply:
I replaced the `get` call with `getOrDefault`.